mflowgen/pad_frame/construct.py

```diff
@@ -0,0 +1,253 @@
+
+  # sr02.2020 b/c now custom_init needs io_file from rtl
+  g.connect_by_name( rtl,         custom_init )
+
+
+  g.connect_by_name( dc,       iflow        )
+  g.connect_by_name( dc,       init         )
+  g.connect_by_name( dc,       power        )
+  g.connect_by_name( dc,       place        )
+  g.connect_by_name( dc,       cts          )
+
+  g.connect_by_name( iflow,    init         )
+  g.connect_by_name( iflow,    power        )
+  g.connect_by_name( iflow,    place        )
+  g.connect_by_name( iflow,    cts          )
+  g.connect_by_name( iflow,    postcts_hold )
+  g.connect_by_name( iflow,    route        )
+  g.connect_by_name( iflow,    postroute    )
+  g.connect_by_name( iflow,    signoff      )
+
+  g.connect_by_name( custom_init,  init     )
+  g.connect_by_name( custom_power, power    )
+
+  # init => init_gdsmerge => init_drc
+  g.connect_by_name( init,         init_gdsmerge )
+  g.connect_by_name( init_gdsmerge,init_drc     )
+
+  g.connect_by_name( init,         power        )
+  g.connect_by_name( power,        place        )
+  g.connect_by_name( place,        cts          )
+  g.connect_by_name( cts,          postcts_hold )
+  g.connect_by_name( postcts_hold, route        )
+  g.connect_by_name( route,        postroute    )
+  g.connect_by_name( postroute,    signoff      )
+  g.connect_by_name( signoff,      gdsmerge     )
+  g.connect_by_name( signoff,      drc          )
+  g.connect_by_name( signoff,      lvs          )
+  g.connect_by_name( gdsmerge,     drc          )
+  g.connect_by_name( gdsmerge,     lvs          )
+
+  g.connect_by_name( signoff,              genlibdb )
+  g.connect_by_name( adk,                  genlibdb )
+#   g.connect_by_name( genlibdb_constraints, genlibdb )
+
+  g.connect_by_name( adk,          pt_signoff   )
+  g.connect_by_name( signoff,      pt_signoff   )
+
+  g.connect_by_name( adk,      debugcalibre )
+  g.connect_by_name( dc,       debugcalibre )
+  g.connect_by_name( iflow,    debugcalibre )
+  g.connect_by_name( signoff,  debugcalibre )
+  g.connect_by_name( drc,      debugcalibre )
+  g.connect_by_name( lvs,      debugcalibre )
+
+  # yes? no?
+  # g.connect_by_name( init_drc,      debugcalibre )
+
+  #-----------------------------------------------------------------------
+  # Parameterize
+  #-----------------------------------------------------------------------
+  g.update_params( parameters )
+
+  # Default order can be found in e.g.
+  # mflowgen/steps/cadence-innovus-init/configure.yml
+  #     - main.tcl
+  #     - quality-of-life.tcl
+  #     - floorplan.tcl
+  #     - pin-assignments.tcl
+  #     - make-path-groups.tcl
+  #     - reporting.tcl
+
+  # I copied this from someone else, maybe glb_top or something
+  # Looks like this order deletes pin assignments and adds endcaps/welltaps
+  # then maybe get clean(er) post-floorplan drc
+  init.update_params(
+    {'order': [
+      'main.tcl','quality-of-life.tcl','floorplan.tcl','io_fillers.tcl'
```

Comment:
None of our other flows remove pin assignments and endcaps/welltaps for the order param. In this case, we removed endcap/welltap because we're not running a flow yet, so adding them created DRCs. And pin assignments were removed because this flow uses an io_file instead of a pin assignment script.

Reply:
I will add your comment as an explanatory note in the code...